codemods/es-get-iterator/index.js

```diff
@@ -0,0 +1,50 @@
+import jscodeshift from 'jscodeshift';
+import { removeImport } from '../shared.js';
+
+/**
+ * @typedef {import('../../types.js').Codemod} Codemod
+ * @typedef {import('../../types.js').CodemodOptions} CodemodOptions
+ */
+
+/**
+ * @param {CodemodOptions} [options]
+ * @returns {Codemod}
+ */
+export default function (options) {
+	return {
+		name: 'es-get-iterator',
+		transform: ({ file }) => {
+			const j = jscodeshift;
+			const root = j(file.source);
+			let dirtyFlag = false;
+
+			const { identifier } = removeImport('es-get-iterator', root, j);
+
+			root
+				.find(j.CallExpression, {
+					callee: {
+						type: 'Identifier',
+						name: identifier,
+					},
+				})
+				.forEach((path) => {
+					const args = path.value.arguments;
+					const newExpression = j.optionalCallExpression(
+						j.memberExpression(
+							// @ts-expect-error
+							args[0],
```

Comment:
```suggestion
					const args = path.value.arguments;
					const arg =
						args[0].type === 'SpreadElement' ? args[0].argument : args[0];

					const newExpression = j.optionalCallExpression(
						j.memberExpression(
							arg,
```

The first arg of `j.memberExpression` needs to be an `ExpressionKind` but `args[0]` has a type of `ExpressionKind | SpreadElement` which is causing the error. In this scenario we can type narrow it and if it is a `SpreadElement` we use the iterable argument they were trying to spread. The argument being a `SpreadElement` is really an edge case but we can cover this case so TypeScript is happy.

The other TS error you are experiencing is for the same reason, so you can use the same logic as here.